tests/generic_config_updater/test_dynamic_acl.py

```diff
@@ -0,0 +1,778 @@
+
+    return forward_packets
+
+
+def generate_drop_packets(setup):
+    """Generate packets with a DST_IP that is never included in our forward rules"""
+
+    drop_packets = {}
+
+    drop_packets["IPV4"] = testutils.simple_tcp_packet(eth_dst=setup["router_mac"],
+                                ip_src=IP_SOURCE,
+                                ip_dst=DST_IP_BLOCKED)
+
+    drop_packets["IPV6"] = testutils.simple_tcpv6_packet(eth_dst=setup["router_mac"],
+                                ipv6_src=IPV6_SOURCE,
+                                ipv6_dst=DST_IPV6_BLOCKED)
+
+    return drop_packets
+
+def build_exp_pkt(input_pkt):
+    """
+    Generate the expected packet for given packet
+    """
+    exp_pkt = Mask(input_pkt)
+    exp_pkt.set_do_not_care_scapy(scapy.Ether, "dst")
+    exp_pkt.set_do_not_care_scapy(scapy.Ether, "src")
+    if input_pkt.haslayer('IP'):
+        exp_pkt.set_do_not_care_scapy(scapy.IP, "ttl")
+        exp_pkt.set_do_not_care_scapy(scapy.IP, "chksum")
+    else:
+        exp_pkt.set_do_not_care_scapy(scapy.IPv6, "hlim")
+
+    return exp_pkt
+
+def expect_acl_table_match_multiple_bindings(duthost, table_name, expected_first_line_content, expected_bindings):
+    """Check if acl table show as expected
+    Acl table with multiple bindings will show as such
+
+    Table_Name  Table_Type  Ethernet4   Table_Description   ingress
+                            Ethernet8
+                            Ethernet12
+                            Ethernet16
+
+    So we must have separate checks for first line and bindings
+    """
+    cmds = "show acl table {}".format(table_name)
+    output = duthost.shell(cmds)
+    pytest_assert(not output['rc'], "'{}' failed with rc={}".format(cmds, output['rc']))
+
+    # Ignore first two lines display. lines less than 3 means no output
+    # Use empty list if no output
+    lines = output['stdout'].splitlines()
+    first_line = [] if len(lines) < 3 else lines[2].split()
+
+    pytest_assert(set(expected_first_line_content) == set(first_line), "ACL table definition doesn't match")
+    bindings = [line.strip() for line in lines[3:]]
+    #Second element of the first line is the first binding
+    bindings.append(first_line[2])
+
+    pytest_assert(set(bindings) == set(expected_bindings))
+
+def expect_acl_rule_match(duthost, rulename, expected_content_list):
+    """Check if acl rule shows as expected"""
+
+    cmds = "show acl rule DYNAMIC_ACL_TABLE {}".format(rulename)
+    output = duthost.shell(cmds)
+    pytest_assert(not output['rc'], "'{}' failed with rc={}".format(cmds, output['rc']))
+
+    lines = output['stdout'].splitlines()
+    actual_list = [] if len(lines) < 3 else lines[2].split()
+    # Ignore the status column
+    expected_len = len(expected_content_list)
+    if len(actual_list) >= expected_len:
+        actual_list = actual_list[0:expected_len]
+
+    pytest_assert(set(expected_content_list) == set(actual_list), "ACL rule does not match!")
```

Comment:
Please also check the status of acl rule. The status should be active.
And there is a helper function show_and_parse. Please try to reuse it so you don''t need to parse the output by yourself.